### PR TITLE
Finish the implementation of MultiplayerColorDefinition and MultiplayerSettings

### LINKED ISSUE #830: thyme.dll!MessageStream::Propagate_Messages() crash
Thyme game crashed in Shell Map Lobby after returning from a Network match with the following callstack:
```
 	game.dat!006d74b7()	Unknown
 	game.dat![Frames below may be incorrect and/or missing, no symbols loaded for game.dat]	Unknown
 	game.dat!004fa2e3()	Unknown
 	game.dat!005abcac()	Unknown
 	game.dat!00403a60()	Unknown
 	game.dat!004fa323()	Unknown
 	game.dat!004fa74b()	Unknown
 	game.dat!00617353()	Unknown
>	thyme.dll!MessageStream::Propagate_Messages() Line 175	C++
 	game.dat!0040fcff()	Unknown
 	thyme.dll!Game_Main(int argc, char * * argv) Line 40	C++
 	thyme.dll!main(int argc, char * * argv) Line 601	C++
 	thyme.dll!Main_Func(HINSTANCE__ * hInstance, HINSTANCE__ * hPrevInstance, char * lpCmdLine, int nCmdShow) Line 625	C++
 	game.dat!008e08ac()	Unknown
 	kernel32.dll!75cc00f9()	Unknown
 	ntdll.dll!77657bbe()	Unknown
 	ntdll.dll!77657b8e()	Unknown
 	game.dat!006e0069()	Unknown
 	game.dat!006e0069()	Unknown
 	game.dat!006e0069()	Unknown
 	game.dat!006e0069()	Unknown
 	game.dat!006e0069()	Unknown
 	game.dat!00650052()	Unknown
 	game.dat!00650052()	Unknown
 	game.dat!006e0069()	Unknown
 	game.dat!006e0069()	Unknown
 	game.dat!006e0069()	Unknown
```

Exception thrown at 0x006D74B7 in game.dat: 0xC0000005: Access violation reading location 0x00000130.

crash

### PR DESCRIPTION
**Pull request by jonwil**
* Fixes #830

### CODE REVIEW

File: src/game/common/multiplayersettings.cpp [hunk 1]
```diff
@@ -31,56 +31,113 @@ const FieldParse MultiplayerColorDefinition::s_colorFieldParsetable[] = {
...
-    m_colorDef1(),
-    m_colorDef2(),
-    m_moneyVec(),
-    m_startingMoney(),
     m_moneyDefault(false)
 {
 }
 
-MultiplayerColorDefinition *MultiplayerSettings::Find_Color_Definition(Utf8String name)
+MultiplayerColorDefinition *MultiplayerSettings::Find_Multiplayer_Color_Definition_By_Name(Utf8String name)
```
Comment (xezon) [inline src/game/common/multiplayersettings.cpp:72]: Not `const Utf8String &name` ?

File: src/game/common/multiplayersettings.cpp [hunk 2]
```diff
@@ -31,56 +31,113 @@ const FieldParse MultiplayerColorDefinition::s_colorFieldParsetable[] = {
...
+        if (i->second.Get_Tooltip_Name() == name) {
+            return &i->second;
+        }
+    }
+
     return nullptr;
 }
 
-MultiplayerColorDefinition *MultiplayerSettings::New_Color_Definition(Utf8String name)
+MultiplayerColorDefinition *MultiplayerSettings::New_Multiplayer_Color_Definition(Utf8String name)
```
Comment (xezon) [inline src/game/common/multiplayersettings.cpp:83]: Not `const Utf8String &name` ?

File: src/game/common/multiplayersettings.cpp [hunk 3]
```diff
@@ -31,56 +31,113 @@ const FieldParse MultiplayerColorDefinition::s_colorFieldParsetable[] = {
...
+    m_numColors = m_colorList.size();
+    return &m_colorList[count];
+}
+
+void MultiplayerSettings::Add_Starting_Money_Choice(const Money &money, bool is_default)
+{
+    m_moneyVec.push_back(money);
+
+    if (is_default) {
+        captainslog_dbgassert(m_moneyDefault == 0, "Cannot have more than one default MultiplayerStartingMoneyChoice");
```
Comment (xezon) [inline src/game/common/multiplayersettings.cpp]: !m_moneyDefault

File: src/game/common/multiplayersettings.h [hunk 4]
```diff
@@ -34,7 +34,13 @@ class MultiplayerColorDefinition
     void Set_Color(RGBColor rgb);
     void Set_Night_Color(RGBColor rgb);
 
-    static void Parse_Color_Definition(INI *ini);
+    static void Parse_Multiplayer_Color_Definition(INI *ini);
+    Utf8String Get_Tooltip_Name() const { return m_tooltipName; }
```
Comment (xezon) [inline src/game/common/multiplayersettings.h:38]: Can return `const Utf8String &`

File: src/game/common/multiplayersettings.h [hunk 5]
```diff
@@ -69,13 +77,65 @@ class MultiplayerSettings : public SubsystemInterface
...
+
+    int Get_Max_Beacons_Per_Player() const { return m_maxBeaconsPerPlayer; }
+    const std::vector<Money> *Get_Money_Vector() const { return &m_moneyVec; }
+    int Get_Start_Countdown_Timer() const { return m_startCountdownTimer; }
+    bool Is_Use_Shroud() const { return m_useShroud; }
+    bool Is_Show_Random_Player_Template() const { return m_showRandomPlayerTemplate; }
+    bool Is_Show_Random_Color() const { return m_showRandomColor; }
+    bool Is_Show_Random_Start_Pos() const { return m_showRandomStartPos; }
+
+    int Get_Num_Colors()
```
Comment (xezon) [inline src/game/common/multiplayersettings.h:106]: Can be `const`

File: src/game/common/multiplayersettings.cpp (hunk 1 above)
Comment (jonwil) [reply src/game/common/multiplayersettings.cpp:72]: no.

File: src/game/common/multiplayersettings.cpp (hunk 2 above)
Comment (jonwil) [reply src/game/common/multiplayersettings.cpp:83]: no.

File: src/game/common/multiplayersettings.cpp (hunk 3 above)
Comment (jonwil) [reply src/game/common/multiplayersettings.cpp]: fixed.

File: src/game/common/multiplayersettings.h (hunk 4 above)
Comment (jonwil) [reply src/game/common/multiplayersettings.h:38]: This is correct to the original.

File: src/game/common/multiplayersettings.h (hunk 5 above)
Comment (jonwil) [reply src/game/common/multiplayersettings.h:106]: It can't be since it changes a member variable.

**Comment by xezon** [comment]
Multiplayer Settings appear to work normally. Tested in Debug configuration. This change also contains a fix for #830, which is unrelated to Multiplayer Settings. Is it ok that unrelated changes are in the same Pull?

**Review by xezon — APPROVED** [review]
Looks good.

**Pull request merged**